gapic/java/java_gapic_pkg.bzl

```diff
@@ -0,0 +1,302 @@
+        "test_deps": attr.label_list(mandatory = False, allow_empty = True),
+        "package_dir": attr.string(mandatory = False),
+        "artifact_group_overrides": attr.string_dict(mandatory = False, allow_empty = True, default = {}),
+        "templates": attr.label_keyed_string_dict(mandatory = False, allow_files = True),
+        "static_substitutions": attr.string_dict(mandatory = False, allow_empty = True, default = {}),
+        "dynamic_substitutions": attr.label_keyed_string_dict(mandatory = False, allow_files = True),
+    },
+    outputs = {"pkg": "%{name}.tar.gz"},
+    implementation = _java_gapic_build_resources_pkg_impl,
+)
+
+def _java_gapic_srcs_pkg_impl(ctx):
+    srcs = []
+    for src_dep in ctx.attr.deps:
+        if is_source_dependency(src_dep):
+            srcs.extend(src_dep.java.source_jars.to_list())
+
+    test_srcs = []
+    for test_src_dep in ctx.attr.test_deps:
+        if is_source_dependency(test_src_dep):
+            test_srcs.extend(test_src_dep.java.source_jars.to_list())
+
+    paths = _construct_package_dir_paths(ctx.attr.package_dir, ctx.outputs.pkg, ctx.label.name)
+
+    # Note the script is more complicated thant it intuitively should be because of limitations
+    # inherent to bazel execution environment: no absolute paths allowed, the generated artifacts
+    # must ensure uniqueness within a build.
+    script = """
+    for src in {srcs}; do
+        mkdir -p {package_dir_path}/src/main/java
+        unzip -q -o $src -d {package_dir_path}/src/main/java
+        rm -r -f {package_dir_path}/src/main/java/META-INF
+    done
+    for test_src in {test_srcs}; do
+        mkdir -p {package_dir_path}/src/test/java
+        unzip -q -o $test_src -d {package_dir_path}/src/test/java
+        rm -r -f {package_dir_path}/src/test/java/META-INF
+    done
+    cd {package_dir_path}
+    tar -zchpf {package_dir}.tar.gz {package_dir_expr}
+    cd -
+    mv {package_dir_path}/{package_dir}.tar.gz {pkg}
+    """.format(
+        srcs = " ".join(["'%s'" % f.path for f in srcs]),
+        test_srcs = " ".join(["'%s'" % f.path for f in test_srcs]),
+        package_dir_path = paths.package_dir_path,
+        package_dir = paths.package_dir,
+        pkg = ctx.outputs.pkg.path,
+        package_dir_expr = paths.package_dir_expr,
+    )
+
+    ctx.actions.run_shell(
+        inputs = srcs + test_srcs,
+        command = script,
+        outputs = [ctx.outputs.pkg],
+    )
+
+java_gapic_srcs_pkg = rule(
+    attrs = {
+        "deps": attr.label_list(mandatory = True, non_empty = True),
+        "test_deps": attr.label_list(mandatory = False, allow_empty = True),
+        "package_dir": attr.string(mandatory = True),
+    },
+    outputs = {"pkg": "%{name}.tar.gz"},
+    implementation = _java_gapic_srcs_pkg_impl,
+)
+
+def java_gapic_proto_gradle_pkg(
+        name,
+        deps,
+        test_deps = None,
+        visibility = None,
+        group = "com.google.api.grpc",
```

Comment:
I'm uneasy about using a google namespace as a default in a build rule that could be used by anyone

Reply:
It is not a build rule, it is a pkg rule, which creates gradle packages. It is easy to not have it there, but it will mean that everybody in googleapis will have to repeatedly set the group. 

Note, that artman also hardcodes that and much deeper: it is hardcoded in .snip files for build gradle files themselves (this was the main reason, why I decided that it is ok to have it like this for now).

Reply:
Well, it's much harder to customize artman. Using Bazel should allow us to dig out these assumptions. I had in mind a Cloud-specific package rule that defaulted this package.

Reply:
Ok, I moved the group/version to the required attributes and googleapis/discovery-artifact-manager repositories will specify them for each artifact